x/tx/signing/aminojson/encoder.go

```diff
@@ -189,3 +194,42 @@ func thresholdStringEncoder(enc *Encoder, msg protoreflect.Message, w io.Writer)
 	_, err = io.WriteString(w, `}`)
 	return err
 }
+
+// sortedObject returns a new object that mirrors the structure of the original
+// but with all maps having their keys sorted.
+func sortedObject(obj interface{}) interface{} {
+	switch v := obj.(type) {
+	case map[string]interface{}:
+		sortedKeys := make([]string, 0, len(v))
+		for key := range v {
+			sortedKeys = append(sortedKeys, key)
+		}
+		sort.Strings(sortedKeys)
+		result := make(map[string]interface{})
+		for _, key := range sortedKeys {
+			result[key] = sortedObject(v[key])
+		}
+		return result
+	case []interface{}:
+		for i, val := range v {
+			v[i] = sortedObject(val)
+		}
+		return v
+	default:
+		return obj
+	}
+}
+
+// sortedJsonStringify returns a JSON with objects sorted by key.
+func sortedJsonStringify(jsonBytes []byte) ([]byte, error) {
+	var obj interface{}
+	if err := json.Unmarshal(jsonBytes, &obj); err != nil {
+		return nil, err
```

Comment:
This should be a JSON validity check, right? So we can return

```
return nil, errors.New("invalid JSON bytes")
```

here and then remove the

```
if !json.Valid(bz) {
			return errors.New("invalid JSON bytes")
		}
```

from cosmosInlineJSON